test/views/agent/partials/RegistrationPartialSpec.scala

```diff
@@ -154,6 +225,48 @@ class RegistrationPartialSpec extends ViewBaseSpec {
           }
         }
       }
+
+      "client has a deregister date in the future, and has partyType of VatGroup" should {
+
+        "display a section for future registration" which {
+
+          lazy val view = registrationPartial(customerDetailsFutureDeregisterOptedOutVatGroup, toLocalDate("2019-01-01"))(messages, mockConfig, user)
+          lazy implicit val document: Document = Jsoup.parse(view.body)
+
+          s"should have the correct title of ${RegistrationPartialMessages.futureDeregisterTitle}" in {
+            elementText("h3") shouldBe RegistrationPartialMessages.futureDeregisterTitle
+          }
+
+          s"have correct content of ${RegistrationPartialMessages.futureDeregisterContent}" in {
+            elementText("p") shouldBe RegistrationPartialMessages.futureDeregisterContent
+          }
+
+          s"link with text of ${RegistrationPartialMessages.futureDeregLink}" in {
+            element("a").text shouldBe RegistrationPartialMessages.futureDeregLink
+          }
+
+          s"link to ${mockConfig.onlineAgentServicesUrl}" in {
+            element("a").attr("href") shouldBe mockConfig.vatHowToRegister
+          }
+        }
+      }
+
+      "agent's client has no patyType" should {
```

Comment:
PartyType

Reply:
We're making burgers here. Gotta have the correct pattys.